spock-spring/boot3-test/src/test/groovy/org/spockframework/boot3/SpringBootTestAnnotationScopedProxyMockSpec.groovy

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2012-2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.boot3
+
+import org.spockframework.boot3.service.HelloWorldService
+import org.spockframework.spring.ScanScopedBeans
+import spock.lang.Specification
+import spock.mock.DetachedMockFactory
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.*
+import org.springframework.context.ApplicationContext
+import org.springframework.context.annotation.*
+import org.springframework.web.context.annotation.RequestScope
+
+/**
+ * Tests enabled scanning of scoped and proxied beans.
```

Comment:
WDYT about testing testing also `UnwrapAopProxyExtension` in Spring 6/Boot 3?

Reply:
Maybe when I figure out a good way to not copy and paste all the tests from `spock-spring/src/test`, but share them across all the spring (boot) projects, while also taking into account what doesn't work cross version.